code/cfile/cfilesystem.cpp

```diff
@@ -1143,6 +1150,24 @@ void cf_build_file_list()
 		}
 	}
 
+#ifndef NDEBUG
+	// if some special/critical files might be shadowed then make sure the user knows about it
+	if ( !critical_shadowed.empty() && !running_unittests ) {
+		SCP_string shadowed;
+		const auto count = critical_shadowed.size();
```

Comment:
Only suggestion I see is that you could use this count variable in the loop range check instead of another `critical_shadowed.size()` call, since you already have it set up anyway.

Reply:
Good point. I changed the loop late during testing to keep the dialog size down and didn't think about the obvious shortcut there.